### PR TITLE
change(components/Modal): Split into Modal and Dialog

### PR DESCRIPTION
**Pull request by kirill-kruchkov**
screen shot 2018-10-01 at 10 18 31

`<Modal>` can be used to display any content as a modal on a backdrop.

### CODE REVIEW

**Comment by zoheiry** [comment]
Approved! just don't forget to add a changelog entry.

File: src/components/Dialog/Dialog_snapshot.spec.js [hunk 1]
```diff
@@ -2,101 +2,99 @@
...
-        <Modal size="normal" open={true}>
-          <Modal.Header>You are changing the status of this order to Shipped</Modal.Header>
-          <Modal.Body>This body has pre-defined paddings.</Modal.Body>
-          <Modal.Footer>
+        <Dialog size="normal" open={true}>
+          <Dialog.Header>You are changing the status of this order to Shipped</Dialog.Header>
+          <Dialog.Body>This body has pre-defined paddings.</Dialog.Body>
+          <Dialog.Footer>
             <Button appearance="primary" onClick={() => {}}>
               Let's do that!
```
Comment (frontalicious-martijn) [inline src/components/Dialog/Dialog_snapshot.spec.js]: Github styling is really bad with `'`. Maybe we can change the content or find a fix for this github rendering.

File: src/components/Modal/Modal.js [hunk 2]
```diff
@@ -1,84 +1,8 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import styled, { css } from 'styled-components';
+import styled from 'styled-components';
 import BaseModal from 'react-modal2';
```
Comment (frontalicious-martijn) [inline src/components/Modal/Modal.js]: Shall we get rid of this dependency and just copy some of the good stuff they implemented? It's not that much.
https://github.com/cloudflare/react-modal2/blob/master/src/Modal.js

File: src/components/Modal/Modal.md [hunk 3]
```diff
@@ -50,114 +38,61 @@ class StyleguideExample extends React.Component {
...
-    </Button>
-    <Button onClick={() => {}}>Cancel</Button>
-  </Modal.Footer>
-</ModalStyleguide>;
-```
+class StyleguideExample extends React.Component {
+  constructor() {
+    super();
+    this.state = { open: false };
+    this.handleOpen = this.handleOpen.bind(this);
```
Comment (frontalicious-martijn) [inline src/components/Modal/Modal.md]: Let's use the es6 notation and get rid of these bindings, also in examples.

File: src/components/Dialog/Dialog.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    padding: ${props => PADDINGS[props.size] || PADDINGS.normal};
+    display: flex;
+    flex-flow: column nowrap;
+    align-items: center;
+
+    > * + * {
+      margin-top: 20px;
+    }
+
+    ${props =>
```
[the remark this answers is not in the document]
Comment (kirill-kruchkov) [reply src/components/Dialog/Dialog.js]: Well we don't do props interpolation inside, so technically we don't need `css` here.

And the formatting doesn't get better:
```
${props =>
    props.size !== 'small'
      && css`
        @media ${DESKTOP} {
          flex-flow: row-reverse wrap;
          align-items: center;
          justify-content: space-between;
          > * + * {
            margin-top: 0;
          }
        }
      `};
  }
```

[the remark this answers is not in the document]
Comment (kirill-kruchkov) [reply src/components/Dialog/Dialog.js]: Fixing this by putting responsiveness-styles to a separate variable

File: src/components/Dialog/Dialog.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    padding: ${props => PADDINGS[props.size] || PADDINGS.normal};
+  }
+`;
+
+/**
+ * Dialog component.
+ */
+function Dialog(props) {
+  const { size, children } = props;
+  return <Root size={size}>{children}</Root>;
```
[the remark this answers is not in the document]
Comment (kirill-kruchkov) [reply src/components/Dialog/Dialog.js]: Renamed to Wrapper to keep things consistent

File: src/components/Modal/Modal.md (hunk 3 above)
Comment (kirill-kruchkov) [reply src/components/Modal/Modal.md]: You can't do that with the current version of Styleguidist

File: src/components/Modal/Modal.js (hunk 2 above)
Comment (kirill-kruchkov) [reply src/components/Modal/Modal.js]: We can do that, but I don't see any reason to spend time on it now. The library is quite untouched for a long time already. If we need to change something - we can easily do that.

File: src/components/Dialog/Dialog_snapshot.spec.js (hunk 1 above)
Comment (kirill-kruchkov) [reply src/components/Dialog/Dialog_snapshot.spec.js]: Changing the wording in examples to avoid that

File: src/components/Modal/Modal.md (hunk 3 above)
Comment (kirill-kruchkov) [reply src/components/Modal/Modal.md]: Just checked. You can't do that with the latest version of styleguidist either

File: src/components/Modal/Modal.js (hunk 2 above)
Comment (kirill-kruchkov) [reply src/components/Modal/Modal.js]: I see one potential benefit of copying: get rid of classnames in favour of styled-components for the sake of uniformity... I'll see what I can do

Comment (kirill-kruchkov) [reply src/components/Modal/Modal.js]: @frontalicious-martijn I did this and also added tests along with the Modal implementation, please take a look when you have time

**Pull request merged**